### PR TITLE
bitnami/kafka - Expose GC metrics via JMX

### PR DESCRIPTION
**Pull request by paulchoi**
**Description of the change**

Expose GC metrics via JMX

GC metrics are categorized under `java.lang.GarageCollector` MBean.
Capture only `CollectionCount` and `CollectionTime` attributes, since
those are the most useful.

**Benefits**

Kafka's garbage collection stats will be exposed via Prometheus

**Possible drawbacks**

It only collects `java.lang.GarbageCollector..*.(CollectionCount|CollectionTime)`.
I find that other `java.lang.GarbageCollector` stats are too noisy and not useful for most users.

**Checklist** 

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [ ] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])

### CODE REVIEW

**Comment by paulchoi** [comment]
VisualVM's view of `java.lang.GarbageCollector`

Screen Shot 2021-11-10 at 11 53 20 AM

Example rendering of avg CollectionTime metric for G1 GC Young Generation.

Screen Shot 2021-11-10 at 11 53 36 AM

File: bitnami/kafka/templates/jmx-configmap.yaml [hunk 1]
```diff
@@ -50,6 +50,10 @@ data:
         name: java_lang_$1_$4_$3_$2
       - pattern: java.lang<type=(.+), name=(.+)><>(\w+)
         name: java_lang_$1_$3_$2
+      - pattern: java.lang<name=(.*), type=GarbageCollector><>CollectionCount
+        type: COUNTER
+      - pattern: java.lang<name=(.*), type=GarbageCollector><>CollectionTime
+        type: GAUGE
```
Comment (juan131) [inline bitnami/kafka/templates/jmx-configmap.yaml:56]: What do you think about adding a generic way to add custom patterns? I mean, adding a new parameter that allows setting your custom patterns in the `values.yaml` so we provide more freedom to users to customize the metrics to be exposed in the JMX exporter.

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks so much for the PR @paulchoi ! We really appreciate it.

Please note that you'll have to bump the version in the **Chart.yaml** according to semver principles.

**Comment by paulchoi** [comment]
> Thanks so much for the PR @paulchoi ! We really appreciate it.
> 
> Please note that you'll have to bump the version in the **Chart.yaml** according to semver principles.

Updated the semver, force-pushed a new commit to my branch. Unfortunately that created a conflict on your end - sorry for the trouble.

File: bitnami/kafka/templates/jmx-configmap.yaml (hunk 1 above)
Comment (paulchoi) [reply bitnami/kafka/templates/jmx-configmap.yaml:56]: Sure, that sounds great. I understand that currently the only other option is to supply a full custom JMX exporter config, which is not awesome.
I'm not much of a Helm guy so I won't be helpful here, but I can make sense of JMX stuff...

**Comment by juan131** [comment]
Don't worry, I just fixed the conflict.

File: bitnami/kafka/templates/jmx-configmap.yaml (hunk 1 above)
Comment (juan131) [reply bitnami/kafka/templates/jmx-configmap.yaml:56]: Can't you extend the current value for `metrics.jmx.config`? I mean, install the chart using the `values.yaml` below:

```yaml
metrics:
  jmx:
    config: |-
      jmxUrl: service:jmx:rmi:///jndi/rmi://127.0.0.1:5555/jmxrmi
      lowercaseOutputName: true
      lowercaseOutputLabelNames: true
      ssl: false
      {{- if .Values.metrics.jmx.whitelistObjectNames }}
      whitelistObjectNames: ["{{ join "\",\"" .Values.metrics.jmx.whitelistObjectNames }}"]
      {{- end }}
      rules:
        - pattern: java.lang<name=(.*), type=GarbageCollector><>CollectionCount
          type: COUNTER
        - pattern: java.lang<name=(.*), type=GarbageCollector><>CollectionTime
          type: GAUGE
```

**Comment by rafariossaa** [comment]
Hi @paulchoi,
  Did you have the chance to take a look to what @juan131 indicated ?

**Pull request closed without merging**